init.lua

```diff
@@ -19,15 +19,38 @@
 -- Thank you, ApolloX, for adding this.
 -- ApollloX: No problem, glade to help.
 
-local player_lights = {}
+goodtorch = {} -- Public API
+goodtorch.modpath = minetest.get_mod_path("goodtorch")
+
+goodtorch.player_lights = {}
+goodtorch.nodes = { -- Replacement nodes (excluding our own nodes, as they don't change)
+	-- Don't need air as air is always air. (That's a lot of air)
+	water = {
+		source = "",
+		flowing = "" -- Uses param2
+	},
+	river = {
+		source = "",
+		flowing = "" -- Uses param2
+	}
+}
+dofile(goodtorch.modpath .. DIR_DELIM .. "api.lua")
+
+if goodtorch.detect_gamemode() == "MTG" then
+	goodtorch.default_nodeset()
+elseif goodtorch.detect_gamemode() == "MCL" then
+	goodtorch.mcl_nodeset()
+else
+	error("This game isn't supported, please go and leave an issue on the github repository with the game name.")
```

Comment:
Right now we just error out when we detect a different gamemode, and won't know till someone makes an issue/report/feature request for that mod/game to be supported.